src/Evolution/Systems/CurvedScalarWave/Worldtube/SingletonActions/IterateAccelerationTerms.cpp

```diff
@@ -25,20 +26,183 @@ void IterateAccelerationTerms::apply(
-  const auto& dilation_factor = get<Tags::TimeDilationFactor>(background);
-  const auto& vel = pos_vel.at(1);
-  const auto self_force_acc =
-      self_force_acceleration(dt_psi_monopole, psi_dipole, vel, charge,
-                              mass.value(), inverse_metric, dilation_factor);
+  auto acc = geodesic_acc;
+  double roll_on = 0.;
+  if (time > turn_on_time.value()) {
+    const auto& particle_position = pos_vel.at(0);
+    const auto& particle_velocity = pos_vel.at(1);
+    const auto& imetric =
+        get<gr::Tags::InverseSpacetimeMetric<double, Dim>>(background);
+    const auto& metric =
+        get<gr::Tags::SpacetimeMetric<double, Dim>>(background);
+    const auto& christoffel =
+        get<gr::Tags::SpacetimeChristoffelSecondKind<double, Dim>>(background);
+    const auto& trace_christoffel =
+        get<gr::Tags::TraceSpacetimeChristoffelSecondKind<double, Dim>>(
+            background);
+    const auto di_imetric =
+        spatial_derivative_inverse_ks_metric(particle_position);
+    const auto dij_imetric =
+        second_spatial_derivative_inverse_ks_metric(particle_position);
+    const auto di_trace_christoffel =
+        spatial_derivative_ks_contracted_christoffel(particle_position);
+    const auto di_metric = spatial_derivative_ks_metric(metric, di_imetric);
+    const auto dij_metric = second_spatial_derivative_metric(
+        metric, di_metric, di_imetric, dij_imetric);
+    const auto di_christoffel = spatial_derivative_christoffel(
+        di_metric, dij_metric, imetric, di_imetric);
+    const auto& u0 = get(get<Tags::TimeDilationFactor>(background));
+
+    // The evolution of the mass is a second order effect so we only include it
+    // starting at the 2nd iteration.
+    const double evolved_mass = iteration == 1
+                                    ? mass.value()
+                                    : mass.value() - charge * get(psi_monopole);
+
+    const auto self_force_acc = self_force_acceleration(
+        dt_psi_monopole, psi_dipole, particle_velocity, charge, evolved_mass,
+        imetric, get<Tags::TimeDilationFactor>(background));
+
+    const double t_minus_turnup = time - turn_on_time.value();
+    roll_on = turn_on_function(t_minus_turnup, turn_on_interval.value());
+    const double dt_roll_on =
+        dt_turn_on_function(t_minus_turnup, turn_on_interval.value());
+    const double dt2_roll_on =
+        dt2_turn_on_function(t_minus_turnup, turn_on_interval.value());
+    for (size_t i = 0; i < Dim; ++i) {
+      acc.get(i) += roll_on * self_force_acc.get(i);
+    }
+    const tnsr::A<double, Dim> u{{u0, get<0>(particle_velocity) * u0,
+                                  get<1>(particle_velocity) * u0,
+                                  get<2>(particle_velocity) * u0}};
+    const tnsr::a<double, Dim> d_psiR{{get(dt_psi_monopole), get<0>(psi_dipole),
+                                       get<1>(psi_dipole), get<2>(psi_dipole)}};
+
+    // Eq.(55)
```

Comment:
In which reference? I know it's stated in the hpp, but it's possible the docs/comments will be updated in one place and not the other and then we won't know what paper to look in. Same for the other times in this cpp